### ISSUE TITLE
2.3.0: external-lib-deps crashes with base

### ISSUE THREAD

**Issue by olafhering** [ocaml/dune#3143]
## Expected Behavior
Should succeed as it did with dune 2.2.0.

## Actual Behavior
````
[   48s] Executing(%build): /bin/sh -e /var/tmp/rpm-tmp.CMlfyp
[   48s] + umask 022
[   48s] + cd /home/abuild/rpmbuild/BUILD
[   48s] + /usr/bin/rm -rf /home/abuild/rpmbuild/BUILDROOT/ocaml-base-0.13.0-22.opt_409.4.x86_64
[   48s] ++ dirname /home/abuild/rpmbuild/BUILDROOT/ocaml-base-0.13.0-22.opt_409.4.x86_64
[   48s] + /usr/bin/mkdir -p /home/abuild/rpmbuild/BUILDROOT
[   48s] + /usr/bin/mkdir /home/abuild/rpmbuild/BUILDROOT/ocaml-base-0.13.0-22.opt_409.4.x86_64
[   48s] + cd ocaml-base-0.13.0
[   48s] + dune_release_pkgs=base
[   48s] + echo 0.13.0
[   48s] + tee VERSION
[   48s] 0.13.0
[   48s] + dune_for_release=
[   48s] + : dune_release_pkgs
[   48s] + test -n base
[   48s] + echo base
[   48s] + dune_for_release=--for-release-of-packages=base
[   48s] + dune installed-libraries
[   48s] bigarray                       (version: OCaml 4.09.0)
[   48s] bytes                          (version: OCaml 4.09.0)
[   48s] compiler-libs                  (version: OCaml 4.09.0)
[   48s] compiler-libs.bytecomp         (version: OCaml 4.09.0)
[   48s] compiler-libs.common           (version: OCaml 4.09.0)
[   48s] compiler-libs.optcomp          (version: OCaml 4.09.0)
[   48s] compiler-libs.toplevel         (version: OCaml 4.09.0)
[   48s] dune-action-plugin             (version: n/a)
[   48s] dune-build-info                (version: n/a)
[   48s] dune-configurator              (version: n/a)
[   48s] dune-glob                      (version: n/a)
[   48s] dune-private-libs.dune-lang    (version: n/a)
[   48s] dune-private-libs.dune_re      (version: n/a)
[   48s] dune-private-libs.ocaml-config (version: n/a)
[   48s] dune-private-libs.stdune       (version: n/a)
[   48s] dune.configurator              (version: n/a)
[   48s] dynlink                        (version: OCaml 4.09.0)
[   48s] raw_spacetime                  (version: OCaml 4.09.0)
[   48s] result                         (version: [distributed with Ocaml])
[   48s] seq                            (version: OCaml 4.09.0)
[   48s] sexplib0                       (version: v0.13.0)
[   48s] stdlib                         (version: OCaml 4.09.0)
[   48s] str                            (version: OCaml 4.09.0)
[   48s] threads                        (version: OCaml 4.09.0)
[   48s] threads.none                   (version: OCaml 4.09.0)
[   48s] threads.posix                  (version: OCaml 4.09.0)
[   48s] uchar                          (version: OCaml 4.09.0)
[   48s] unix                           (version: OCaml 4.09.0)
[   48s] + dune external-lib-deps --for-release-of-packages=base @install
[   48s] Internal error, please report upstream including the contents of _build/log.
[   48s] Description:
[   48s] ("Map.find_exn: failed to find key",
[   48s] { key = "core_kernel.base_for_tests"; keys = [ "core_kernel" ] })
[   48s] Backtrace:
[   48s] Raised at file "src/stdune/code_error.ml", line 9, characters 30-62
[   48s] Called from file "src/dune/findlib/findlib.ml", line 498, characters 8-44
[   48s] Called from file "src/dune/lib.ml", line 1843, characters 24-59
[   48s] Called from file "src/dune/lib.ml", line 1189, characters 10-25
[   48s] Called from file "src/dune/lib.ml", line 1203, characters 21-49
[   48s] Called from file "src/dune/lib.ml", line 1203, characters 21-49
[   48s] Called from file "src/dune/lib.ml", line 1181, characters 10-38
[   48s] Called from file "src/dune/lib.ml", line 1287, characters 25-76
[   48s] Called from file "list.ml", line 121, characters 24-34
[   48s] Called from file "src/dune/lib.ml", line 1269, characters 6-1023
[   48s] Called from file "src/dune/lib.ml" (inlined), line 1110, characters 9-75
[   48s] Called from file "src/dune/lib.ml", line 1109, characters 6-104
[   48s] Called from file "src/dune/lib.ml", line 1860, characters 10-61
[   48s] Called from file "src/dune/odoc.ml", line 640, characters 17-104
[   48s] Called from file "src/stdune/list.ml", line 19, characters 12-15
[   48s] Called from file "list.ml", line 103, characters 22-25
[   48s] Called from file "src/stdune/list.ml" (inlined), line 5, characters 19-33
[   48s] Called from file "src/stdune/list.ml", line 40, characters 29-39
[   48s] Called from file "src/dune/odoc.ml" (inlined), line 633, characters 7-538
[   48s] Called from file "src/dune/odoc.ml", line 632, characters 6-552
[   48s] Called from file "map.ml", line 295, characters 20-25
[   48s] Called from file "src/stdune/exn.ml", line 13, characters 8-11
[   48s] Re-raised at file "src/stdune/exn.ml", line 19, characters 4-11
[   48s] Called from file "src/memo/implicit_output.ml", line 120, characters 4-162
[   48s] Called from file "src/dune/rules.ml" (inlined), line 192, characters 20-71
[   48s] Called from file "src/dune/rules.ml", line 195, characters 20-33
[   48s] Called from file "src/dune/build_system.ml", line 356, characters 23-56
[   48s] Called from file "src/dune/gen_rules.ml", line 429, characters 2-387
[   48s] Called from file "src/fiber/fiber.ml", line 196, characters 36-41
[   48s] Called from file "src/fiber/fiber.ml", line 102, characters 8-15
[   48s] 
[   48s] I must not segfault.  Uncertainty is the mind-killer.  Exceptions are
[   48s] the little-death that brings total obliteration.  I will fully express
[   48s] my cases.  Execution will pass over me and through me.  And when it
[   48s] has gone past, I will unwind the stack along its path.  Where the
[   48s] cases are handled there will be nothing.  Only I will remain.
[   48s] error: Bad exit status from /var/tmp/rpm-tmp.CMlfyp (%build)
[   48s] 
[   48s] 
[   48s] RPM build errors:
[   48s]     Bad exit status from /var/tmp/rpm-tmp.CMlfyp (%build)
[   48s] 
[   48s] sheep68 failed "build ocaml-base.spec" at Sat Feb 15 15:17:28 UTC 2020.
````

## Specifications

- Version of `dune` (output of `dune --version`):
2.3.0, commit 3bbbc974cc1802249e12779cda24cdaf857c9368 and f3319aacb8ee7968b6babadae4c419f10df5964e
- Version of `ocaml` (output of `ocamlc --version`)
4.07, 4.08, 4.09, 4.10 branches from ocaml.git
- Operating system (distribution and version):
Leap

**Comment by olafhering** [comment]
Oh, does it mean there is now a cycle `dune` -> `core_kernel` -> `dune`?!

**Comment by rgrinberg** [comment]
What dune command are you running?

**Comment by olafhering** [comment]
> What dune command are you running?

`[   48s] + dune external-lib-deps --for-release-of-packages=base @install`

**Comment by olafhering** [comment]
This is fixed with the moved 2.3.0 tag.

**Comment by olafhering** [comment]
Thank you for the quick fix!
